rmgpy/molecule/parser/inchi.py

```diff
@@ -0,0 +1,109 @@
+import re
+
+INCHI_PREFIX = 'InChI=1'
+
+"""
+
+The prefix with the multiplicity information.
+
+For example, a triplet biradical will bear the 
+following multiplicity layer in the augmented InChI:
+
+InChI=1/.../mult3
+
+The singlet equivalent of aforementioned biradical has the following
+augmented InChI:
+
+InChI=1/.../mult1
+
+"""
+MULT_PREFIX = '/mult'
+
+
+
+"""
+The prefix with the information on the distribution of unpaired electrons across the atoms.
+
+For example, a triplet biradical with unpaired electrons on atom 1 and atom 3 
+will have the following unpaired electron layer in the augmented InChI:
+
+InChI=1/.../mult3/u1,3
+
+The indices refer to the 1-based indices in the InChI string (NOT the 0-based
+    indices of the Molecule container!)
+
+"""
+U_LAYER_PREFIX = '/u'
+
+"""The separator that separates the indices of the atoms that bear unpaired electrons."""
+U_LAYER_SEPARATOR = ','
+
+def decompose(string):
+    """
+    Converts an augmented inchi into an inchi, multiplicity and indices array for the atoms
+    bearing unpaired electrons.
+
+    returns: 
+    - multiplicity (int)
+    - indices array for the atoms bearing unpaired electrons.
+
+    """
+
+    if not MULT_PREFIX in string:
+        return string, None, []
+
+    inchi, mult_ulayer = string.split(MULT_PREFIX)
+    
+    if not U_LAYER_PREFIX in mult_ulayer:
+        mult = int(mult_ulayer)
+        u_indices = []    
+        return inchi, mult, u_indices
+
+    mult, ulayer = mult_ulayer.split(U_LAYER_PREFIX)
+    mult = int(mult)
+    u_indices = [int(i) for i in ulayer.split(U_LAYER_SEPARATOR)]
+    return inchi, mult, u_indices
+
+def ignore_prefix(string):
+    """
+    Splits off the 'InChI=1S' or 'InChI=1' layer of an InChI
+    and returns the last part.
+    """
+
+    assert INCHI_PREFIX in string, 'Not a valid InChI: {}'.format(string)
```

Comment:
Properly we should raise an exception like `InchiException` rather than `AssertionError` to enable smarter error handling downstream (eg. the website may want to implement a `Try ... Except` block for user mistakes.)   Ideally assertions should only check for bugs in the code, not mistakes in the input or chemistry.   (Though I often lazily use them for this too  :wink:)